plugins/ingest/src/main/java/org/elasticsearch/ingest/Data.java

```diff
@@ -129,4 +129,33 @@ public String getId() {
     public boolean isModified() {
         return modified;
     }
+
+    public Map<String, Object> asMap() {
```

Comment:
makes sense, although it depends on what you do with the Streamable aspect :) if you do add an enhanced Data class that one might as well implement ToXContent, should be an easy change.